src/prefect/engine/result/base.py

```diff
@@ -120,13 +118,7 @@ def store_safe_value(self) -> None:
         if self.value is None:
             return
         if self.safe_value == NoResult:
-            assert isinstance(
-                self.result_handler, ResultHandler
-            ), "Result has no ResultHandler"  # mypy assert
-            value = self.result_handler.write(self.value)
-            self.safe_value = SafeResult(
-                value=value, result_handler=self.result_handler
-            )
+            self.safe_value = SafeResult(value=self.filepath)
```

Comment:
So far I've left the SafeResult stuff still in here, though I've unhooked what uses it (the serializer's `validation_fn=get_safe` I delete below), so need to consider if any calls of `store_safe_value` should still exist.

Reply:
`store_safe_value` should be equivalent to `write` so I think we we can remove all those calls.  As discussed offline, we do need to keep the `SafeResult` class itself for backwards compatibility of deserialization logic.

Reply:
Ok ripped most everything out re: store_safe_result in 2125c24